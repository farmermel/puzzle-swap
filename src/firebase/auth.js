import { auth } from './firebase';

export const doCreateUserWithEmailAndPassword = (email, password) =>
  auth.createUserWithEmailAndPassword(email, password);

export const doSignInWithEmailAndPassword = (email, password) => 
  auth.signInWithEmailAndPassword(email, password);

export const onAuthStateChanged = user => 
  auth.onAuthStateChanged(user);

export const doSignOut = () => 
  auth.signOut();