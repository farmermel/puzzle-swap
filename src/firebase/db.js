import { db } from './firebase';

export const doCreateUser = (id, username, email) => 
  db.ref(`users/${id}`).set({
    username,
    email
  });

// export const onceGetUsers = () => 
//   db.ref('users').once('value');

export const getOnce = path =>
  db.ref(path).once('value');

export const getFirebaseKey = childNode => 
  db.ref().child(childNode).push().key;

export const postUpdate = updates => 
  db.ref().update(updates);

export const watchData = watchNode => 
  db.ref(watchNode);

// export const getData = watchNode => 
//   watchData(watchNode).on('value', (snapshot) => {
//     return snapshot.val()
//   })