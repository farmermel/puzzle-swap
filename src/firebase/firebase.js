import * as firebase from 'firebase';
import { firebaseKey,
  fbAuthDomain,
  fbdbUrl,
  fbProjId,
  fbMessagingSenderId } from '../helpers/apiKeys';

const config = {
  apiKey: firebaseKey,
  authDomain: fbAuthDomain,
  databaseURL: fbdbUrl,
  projectId: fbProjId,
  storageBucket: "",
  messagingSenderId: fbMessagingSenderId
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

const db = firebase.database();

export {
  db,
  auth,
};