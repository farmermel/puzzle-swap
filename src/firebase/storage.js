import { storage } from './firebase';

export const getStoreRef = path =>
  storage.ref().child(path);

export const putInStore = (ref, blob) =>
  ref.put(blob);