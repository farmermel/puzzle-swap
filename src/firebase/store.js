import { store } from './firebase';

export const getStoreRef = path =>
  store.ref().child(path)

export const putInStore = (ref, blob) =>
  ref.put(blob)