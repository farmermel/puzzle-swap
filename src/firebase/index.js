import * as auth from './auth';
import * as db from './db';
import * as firebase from './firebase';
import * as store from './store';

export {
  auth,
  db,
  store,
  firebase,
};