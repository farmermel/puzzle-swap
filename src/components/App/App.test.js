import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { auth } from '../../firebase';
import { App,
mapStateToProps,
mapDispatchToProps } from './App';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App setUser={jest.fn()}
                           user={null} />, 
                      {disableLifecycleMethods: true});
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('componentDidMount', () => {
    it.skip('calls onAuthStateChanged method from auth and sets user if user exists', () => {
      // const mockUser = { uid: 7};
      // auth.onAuthStateChanged = jest.fn().mockImplementation((mockUser) => {
      //   mockUser ? setUser(mockUser) : setUser(null)
      // })
      // wrapper.instance().props.user = mockUser;
      // wrapper.instance().componentDidMount();
      // expect(wrapper.instance().props.setUser).toHaveBeenCalled();
    })

    it.skip('calls onAuthStateChanged method from auth and sets user as null if user doesn\'t exist', () => {

    })
  })

  describe('mapStateToProps', () => {
    it('maps state to props', () => {
      const mockStore = {
        user: { uid: 7}
      }
      const mapped = mapStateToProps(mockStore);
      expect(mapped.user).toEqual(mockStore.user);
    })
  })

  describe('mapDispatchToProps', () => {
    it('maps dispatch to props', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      expect(mockDispatch).not.toHaveBeenCalled();
      mapped.setUser();
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
});