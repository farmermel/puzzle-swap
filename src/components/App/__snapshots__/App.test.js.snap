// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    setUser={[Function]}
    user={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          user={null}
/>,
        <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    path="/post-puzzle-form"
                    render={[Function]}
          />
          <Route
                    path="/login"
                    render={[Function]}
          />
          <Route
                    path="/sign-up"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    path="/messages"
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "user": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              path="/post-puzzle-form"
              render={[Function]}
/>,
            <Route
              path="/login"
              render={[Function]}
/>,
            <Route
              path="/sign-up"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              path="/messages"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/post-puzzle-form",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/login",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/sign-up",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/messages",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            user={null}
/>,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        path="/post-puzzle-form"
                        render={[Function]}
            />
            <Route
                        path="/login"
                        render={[Function]}
            />
            <Route
                        path="/sign-up"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        path="/messages"
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "user": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                path="/post-puzzle-form"
                render={[Function]}
/>,
              <Route
                path="/login"
                render={[Function]}
/>,
              <Route
                path="/sign-up"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                path="/messages"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/post-puzzle-form",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/login",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/sign-up",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/messages",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
